packages/faceted-search/src/components/AddFacetPopover/AddFacetPopover.component.js

```diff
@@ -37,55 +69,189 @@ AddFacetRow.propTypes = {
+	const [filterValue, setFilterValue] = useState(initialFilterValue || '')
+
+	const categories = useMemo(
+		() =>
+			badgesDefinitions
+				.filter(badgeDefinition => !!badgeDefinition.metadata.category)
+				.map(badgeDefinition => badgeDefinition.metadata.category)
+				.filter((categoryName, index, arr) => arr.indexOf(categoryName) === index),
+		[badgesDefinitions],
+	);
+
+	const screensCount = categories.length + 1;
+	const [screensHeight, setScreensHeight] = useState(times(screensCount, () => 0));
+	const screensRef = useRef(times(screensCount, createRef));
+
+	useEffect(() => {
+		setScreensHeight(times(screensCount, index => screensRef.current[index].current.clientHeight));
+	}, []);
+
 	const onFilter = (_, value) => {
 		setFilterValue(value);
 	};
 	const resetFilter = () => setFilterValue('');
-	const badgesDefinitionsFaceted = useMemo(
+
+	const onCategoryChange = newCategory => {
+		setCategory(newCategory);
+		resetFilter();
+	};
+
+	const badgesWithoutCategory = useMemo(
 		() =>
-			badgesDefinitions.filter(filterByLabel(filterValue.toLowerCase().trim())).sort(sortByLabel),
-		[badgesDefinitions, filterValue],
+			badgesDefinitions.filter(badgeDefinition => !badgeDefinition.metadata.category),
+		[badgesDefinitions],
 	);
-	const addFacetId = `${id}-add-facet-popover`;
+
+	const screens = [
+		{
+			category: null,
+			rows: useMemo(
+				() =>
+					[...categories, ...badgesWithoutCategory]
+						.sort(sortByLabel)
+						.filter(filterByLabel(filterValue.toLowerCase().trim())),
+				[badgesDefinitions, filterValue]
+			),
+		},
+		...categories.map(categoryName => ({
+			category: categoryName,
+			rows: useMemo(
+				() => badgesDefinitions
+					.filter(badgeDefinition => badgeDefinition.metadata.category === categoryName)
+					.filter(filterByLabel(filterValue.toLowerCase().trim()))
+					.sort(sortByLabel),
+				[filterValue]
+			),
+		})),
+	];
+
+	const categoryScreenIndex = screens.findIndex(screen => screen.category === category);
+
 	return (
 		<div id={addFacetId} className={theme('tc-add-facet-popover')}>
-			<RichLayout.Body id={addFacetId}>
-				<FilterBar
-					autoFocus={false}
-					className={theme('tc-add-facet-popover-filter')}
-					dockable={false}
-					docked={false}
-					iconAlwaysVisible
-					id={`${addFacetId}-filter`}
-					placeholder={t('ADD_FACET_FILTER_PLACEHOLDER', {
-						defaultValue: 'Find a filter',
-					})}
-					onToggle={resetFilter}
-					onFilter={onFilter}
-					value={filterValue}
-				/>
-				<div className={theme('tc-add-facet-popover-row-container')}>
-					{badgesDefinitionsFaceted.map(badgeDefinition => (
-						<AddFacetRow
-							badgeDefinition={badgeDefinition}
-							id={addFacetId}
-							key={badgeDefinition.properties.label}
-							label={badgeDefinition.properties.label}
-							onClick={onClick}
+			<div
+				id={addFacetId}
+				className={theme('tc-add-facet-popover-container')}
+				style={{
+					height: screensHeight[categoryScreenIndex],
```

Comment:
why height & maxHeight with the same value?

Reply:
You're right, fix that